dev/docker/kerberos-hive/start.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+#
+# Copyright 2024 Datastrato Pvt Ltd.
+# This software is licensed under the Apache License version 2.
+#
+
+# start ssh
+HOSTNAME=`hostname`
+service ssh start
+ssh-keyscan localhost > /root/.ssh/known_hosts
+ssh-keyscan 0.0.0.0 >> /root/.ssh/known_hosts
+
+# init the Kerberos database
+echo -e "${PASS}\n${PASS}" | kdb5_util create -s
+
+# start Kerberos related service
+service krb5-kdc start
+service krb5-admin-server start
+
+# create Kerberos principal and keytab files
+FQDN="HADOOPKRB"
+ADMIN="admin"
+PASS="Admin12!"
+KRB5_KTNAME=/etc/admin.keytab
+
+echo -e "${PASS}\n${PASS}" | kadmin.local -q "addprinc ${ADMIN}/admin"
+echo -e "${PASS}\n${PASS}" | kadmin.local -q "addprinc cli@${FQDN}"
+echo -e "${PASS}\n${PASS}" | kadmin.local -q "addprinc hdfs/${HOSTNAME}@${FQDN}"
+echo -e "${PASS}\n${PASS}" | kadmin.local -q "addprinc HTTP/${HOSTNAME}@${FQDN}"
+echo -e "${PASS}\n${PASS}" | kadmin.local -q "addprinc hive/${HOSTNAME}@${FQDN}"
+kadmin.local -q "ktadd -norandkey -k ${KRB5_KTNAME} cli@${FQDN}"
+kadmin.local -q "ktadd -norandkey -k ${KRB5_KTNAME} hdfs/${HOSTNAME}@${FQDN}"
+kadmin.local -q "ktadd -norandkey -k ${KRB5_KTNAME} HTTP/${HOSTNAME}@${FQDN}"
+
+kadmin.local -q "xst -k /hdfs.keytab -norandkey hdfs/${HOSTNAME}@${FQDN}"
+kadmin.local -q "xst -k /hdfs.keytab -norandkey HTTP/${HOSTNAME}@${FQDN}"
+
+kadmin.local -q "ktadd -norandkey -k ${KRB5_KTNAME} hive/${HOSTNAME}@${FQDN}"
+kadmin.local -q "xst -k /hive.keytab -norandkey hive/${HOSTNAME}@${FQDN}"
+kadmin.local -q "xst -k /cli.keytab -norandkey cli@${FQDN}"
+
+echo -e "${PASS}\n" | kinit hive/${HOSTNAME}
+
+# Update the configuration file
+sed -i "s/mockhost/${HOSTNAME}/g" ${HADOOP_CONF_DIR}/hdfs-site.xml
+sed -i "s/mockhost/${HOSTNAME}/g" ${HADOOP_CONF_DIR}/core-site.xml
+sed -i "s/mockhost/${HOSTNAME}/g" ${HIVE_HOME}/conf/hive-site.xml
+
+# format HDFS
+${HADOOP_HOME}/bin/hdfs namenode -format -nonInteractive
+
+echo "Starting HDFS..."
+echo "Starting NameNode..."
+${HADOOP_HOME}/sbin/hadoop-daemon.sh start namenode
+
+echo "Starting DataNode..."
+${HADOOP_HOME}/sbin/start-secure-dns.sh
```

Comment:
Is this the command to start DN?

Reply:
Ping me when everything is worked and code is polished/ready-to-review.

Reply:
> Is this the command to start DN?

Yes,  Hive with Kerberos enable is different with that without Kerberos, If we use `${HADOOP_HOME}/sbin/hadoop-daemon.sh start datanode`, the following error will occur:
```
JSVC_HOME is not set correctly so jsvc cannot be found. jsvc is required to run secure datanodes
```
That's why Qe Qi change it.